rust/navigator/src/navstate.rs

```diff
@@ -610,954 +602,1305 @@ impl State {
+                }
+                Screen::Scan => ScreenData::Scan,
                 Screen::Transaction(ref t) => {
-                    match t.action() {
-                        transaction_parsing::Action::Derivations{content, network_info, checksum: _, network_specs_key: _} => format!("\"content\":{{{}}},\"network_info\":{{{}}},\"type\":\"import_derivations\"", content, network_info),
-                        transaction_parsing::Action::Sign{content, checksum: _, has_pwd: _, author_info, network_info} => {
-                            let action_type = match new_navstate.modal {
-                                Modal::SignatureReady(_) => "done",
-                                _ => "sign",
-                            };
-                            format!("\"content\":{{{}}},\"author_info\":{{{}}},\"network_info\":{{{}}},\"type\":\"{}\"", content, author_info, network_info, action_type)
+                    let (content, ttype, author_info, network_info) = match t.action() {
+                        TransactionAction::Derivations {
+                            content,
+                            network_info,
+                            ..
+                        } => (
+                            content,
+                            TransactionType::ImportDerivations,
+                            None,
+                            Some(network_info),
+                        ),
+                        TransactionAction::Sign {
+                            content,
+                            author_info,
+                            network_info,
+                            ..
+                        } => (
+                            content,
+                            TransactionType::Sign,
+                            Some(author_info),
+                            Some(network_info),
+                        ),
+                        TransactionAction::Stub { s, .. } => (s, TransactionType::Stub, None, None),
+                        TransactionAction::Read { r } => (r, TransactionType::Read, None, None),
+                    };
+                    ScreenData::Transaction {
+                        f: MTransaction {
+                            content,
+                            ttype,
+                            author_info,
+                            network_info: network_info.map(|i| MSCNetworkInfo {
+                                network_title: i.title,
+                                network_logo: i.logo,
+                            }),
                         },
-                        transaction_parsing::Action::Stub(content, _, _) => format!("\"content\":{{{}}},\"type\":\"stub\"", content),
-                        transaction_parsing::Action::Read(content) => format!("\"content\":{{{}}},\"type\":\"read\"", content),
                     }
-                },
+                }
                 Screen::SeedSelector | Screen::SelectSeedForBackup => {
```

Comment:
These should be two disctinct screens, as they have different actions. They will probably also have different UI later.